urllib3/contrib/pyopenssl.py

```diff
@@ -127,6 +129,17 @@ def extract_from_urllib3():
     util.ssl_.IS_PYOPENSSL = False
 
 
+def _validate_dependencies_met():
+    """
+    Verifies that PyOpenSSL's package-level dependencies have been met.
+    Throws `ImportError` if they are not met.
+    """
+    # Method added in `cryptography==1.1`; not available in older versions
+    from cryptography.x509.extensions import Extensions
+    if not hasattr(Extensions, "get_extension_for_class"):
+        raise ImportError("'cryptography' module missing required function (try upgrading the module?): 'cryptography.x509.extensions.Extensions.get_extension_for_class'")
```

Comment:
Regarding validating the 1.2-1.3.3 range, actually, does anyone know why that requirement was put into place?  I could easily just validate the version-string directly if that's the preference.  But I feel like the spirit of duck typing would want this test to check for the particular functionality in that release urllib3 depends on.